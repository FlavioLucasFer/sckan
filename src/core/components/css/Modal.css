.modal-content>* {
	color: #000;
}

.modal-footer {
	line-height: 50px !important;
	padding: 0 !important;
}

.footer-button {
	margin: 0px 5px !important;
}