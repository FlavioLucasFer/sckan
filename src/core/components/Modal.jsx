import './css/Modal.css';
import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

Modal.propTypes = {
	title: PropTypes.string.isRequired,
	className: PropTypes.string,
	contentClassName: PropTypes.string,
	actions: PropTypes.arrayOf(Object),
	children: PropTypes.element,
};

function Modal(props) {
	const {
		title,
		className,
		contentClassName,
		actions,
		children,
		...rest
	} = props;

	useEffect(() => {
		M.Modal.init(document.querySelectorAll('.modal'), {
			dismissible: false,
		});
	}, []);

	function renderFooter() {
		if (!actions) {
			return;
		}
		
		return (
			<div className="modal-footer">
				{actions.map((e, i) => (
					<a {...e} 
						key={i} 
						className={`waves-effect ${e.className || ''} btn-flat footer-button`}>
							{e.title}
					</a>
				))}
			</div>
		);
	}

	return (
		<div {...rest}
			className={`modal ${className || ''} ${actions ? 'modal-fixed-footer' : ''}`}>
			<div className={`modal-content ${contentClassName || ''}`}>
				{title ? 
					<h5>{title}</h5>
				:
					null
				}
				{children}
			</div>
			{renderFooter()}
		</div>
	);
}

export default Modal;
