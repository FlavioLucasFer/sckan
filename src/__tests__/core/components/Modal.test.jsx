import React from 'react';
import { shallow } from 'enzyme';

import Modal from 'core/components/Modal';

describe('Testing Modal core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Modal title="modal test">
				<h2>testing</h2>
			</Modal>
		);

		expect(wrapper.debug()).toMatchSnapshot();

		const actions = [
			{ title: 'action1', className: 'red', href: 'hreftest1' },
			{ title: 'action2', className: 'green', href: 'hreftest2' },
			{ title: 'action3', className: 'teal', href: 'hreftest3' },
		];

		const options = {
			opacity: 0.9,
			inDuration: 750,
			onOpenStart: () => console.log('testing modal options'),
		};

		wrapper.setProps({
			title: 'testing modal',
			className: 'teal',
			actions,
			options,
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});