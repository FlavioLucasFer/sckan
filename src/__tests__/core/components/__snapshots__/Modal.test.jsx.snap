// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal core component should render correctly 1`] = `
"<div className=\\"modal  \\">
  <div className=\\"modal-content\\">
    <h5>
      modal test
    </h5>
    <h2>
      testing
    </h2>
  </div>
</div>"
`;

exports[`Testing Modal core component should render correctly 2`] = `
"<div className=\\"modal teal modal-fixed-footer\\">
  <div className=\\"modal-content\\">
    <h5>
      testing modal
    </h5>
    <h2>
      testing
    </h2>
  </div>
  <div className=\\"modal-footer\\">
    <a title=\\"action1\\" className=\\"waves-effect red btn-flat footer-button\\" href=\\"hreftest1\\">
      action1
    </a>
    <a title=\\"action2\\" className=\\"waves-effect green btn-flat footer-button\\" href=\\"hreftest2\\">
      action2
    </a>
    <a title=\\"action3\\" className=\\"waves-effect teal btn-flat footer-button\\" href=\\"hreftest3\\">
      action3
    </a>
  </div>
</div>"
`;
